app/services/follow_service.rb

```diff
@@ -31,6 +33,8 @@ def remove_reason(entity, user, reason)
     protected
 
     def should_auto_follow(entity, user, reason)
+      follow = Follow.where(followed: entity, user: user).first
+      return false if follow && follow.muted
```

Comment:
This prevents the auto-follow from happening, when you've manually unfollowed it